zetta_utils/training/lightning/train.py

```diff
@@ -58,7 +69,91 @@ def lightning_train(
         Must be a full training state checkpoint created by PytorchLightning rather
         than a model checkpoint. If ``full_state_ckpt_path=="last"``, the latest
         checkpoint for the given experiment will be identified and loaded.
+    :param num_nodes: Number of GPU nodes for distributed training.
+    :param nproc_per_node: Number of GPU workers per node.
+    :param retry_count: Max retry count for the master train job;
+        excludes failures due to pod distruptions.
+    :param local_run: If True run the training locally.
+    :param follow_logs: If True, eagerly print logs from the pod.
+        If False, will wait until job completes successfully.
+    :param image: Container image to use.
+    :param cluster_name: Cluster configuration.
+    :param cluster_region: Cluster configuration.
+    :param cluster_project: Cluster configuration.
+    :param env_vars: Custom env variables to be set on pods.
+    :param resource_limits: K8s reource limits per pod.
+    :param resource_requests: K8s resource requests per pod.
     """
+
+    if local_run:
+        _lightning_train(regime, trainer, train_dataloader, val_dataloader=val_dataloader)
+        return
+
+    assert image is not None, "Must provide a container image for remote training."
+    execution_id = mazepa.id_generation.get_unique_id(
+        prefix="exec", slug_len=4, add_uuid=False, max_len=50
+    )
+
+    cluster_info = resource_allocation.k8s.parse_cluster_info(
+        cluster_name=cluster_name,
+        cluster_region=cluster_region,
+        cluster_project=cluster_project,
+    )
+
+    train_spec = {
+        "@type": "lightning_train",
+        "regime": regime,
```

Comment:
should prolly use [[builder lookup](https://github.com/ZettaAI/zetta_utils/blob/main/zetta_utils/builder/build.py#L26) here